### PR TITLE
Mapper: simplify constructor

### PR DESCRIPTION
**Pull request by enumag**

### CODE REVIEW

**Comment by JanTvrdik** [comment]
Reasoning? BC impact?

**Comment by enumag** [comment]
`Nette\Utils\Callback::check()` just checks if the argument is callable and throws an exception otherwise which is pretty much the same as what callable type hint does and since we require 5.4 now we can simplify it. Unless someone was catching that exception (which I highly doubt) there should be no BC break.

File: src/Iterators/Mapper.php [hunk 1]
```diff
@@ -19,10 +19,10 @@ class Mapper extends \IteratorIterator
 	private $callback;
 
 
-	public function __construct(\Traversable $iterator, $callback)
+	public function __construct(\Traversable $iterator, callable $callback)
```
Comment (fprochazka) [inline src/Iterators/Mapper.php]: http://phpfashion.com/php-5-4-ma-nepouzitelny-typ-callable

Comment (enumag) [inline src/Iterators/Mapper.php]: Yes I read that. Irrelevant in this case.

**Comment by dg** [comment]
what about `Mapper::__construct added typehint callable`?

**Comment by enumag** [comment]
@dg Done.

@fprochazka If I understand the article correctly, it basically says to use `Nette\Utils\Callback::check($callback, TRUE);` instead of `Nette\Utils\Callback::check($callback, TRUE);` or callable typehint. If @dg wants that instead I don't mind.

**Pull request merged**